src/sails_forth/datomic.clj

```diff
@@ -0,0 +1,202 @@
+   {:db/ident (field-ident ns-prefix "helptext")
+    :db/doc "Salesforce field help text"
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/one}
+
+   {:db/ident (compound-ident ns-prefix "address" "street")
+    :db/doc "Salesforce address street"
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/one}
+   {:db/ident (compound-ident ns-prefix "address" "city")
+    :db/doc "Salesforce address city"
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/one}
+   {:db/ident (compound-ident ns-prefix "address" "state-code")
+    :db/doc "Salesforce address state"
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/one}
+   {:db/ident (compound-ident ns-prefix "address" "postal-code")
+    :db/doc "Salesforce address postal code"
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/one}
+   {:db/ident (compound-ident ns-prefix "address" "country-code")
+    :db/doc "Salesforce address country"
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/one}])
+
+(defn object-schema
+  [ns-prefix object-key fields]
+  (letfn [(field-datoms [[key field]]
+            (let [{:keys [name
+                          label
+                          type
+                          calculatedFormula
+                          inlineHelpText
+                          unique]}
+                  field
+                  cardinality (if (= "multipicklist" type)
+                                :db.cardinality/many
+                                :db.cardinality/one)
+                  recordtype? (= key :recordtype)
+                  valuetype (cond
+                              recordtype? :db.type/string
+                              (= "double" type)
+                              (case (clj/double-type field)
+                                :long :db.type/long
+                                :bigint :db.type/bigint
+                                :bigdec :db.type/bigdec)
+                              :else
+                              (get datomic-types type))]
+              (cond-> {:db/ident (field-attr ns-prefix object-key key)
+                       :db/doc label
+                       :db/valueType valuetype
+                       :db/cardinality cardinality
+                       (field-ident ns-prefix "name") name
+                       (field-ident ns-prefix "type") type}
+                ;; Sorta funny that id types don't have :unique true
+                (or (= "id" type) unique)
+                (assoc :db/unique :db.unique/identity)
+                (= "address" type)
+                (assoc :db/isComponent true)
+                calculatedFormula
+                (assoc (field-ident ns-prefix "formula") calculatedFormula)
+                inlineHelpText
+                (assoc (field-ident ns-prefix "helptext") inlineHelpText))))]
+    (into [] (map field-datoms) fields)))
+
+(defn build-schema!
```

Comment:
does this actually have side effects? is the `!` meant to capture possible client interactions in `c/get-fields`?